File: app/src/main/java/com/bsas/androiddevs/manejoerrores/repository/db/dao/MovieDao.java
package com.bsas.androiddevs.manejoerrores.repository.db.dao;

import androidx.room.Dao;
import androidx.room.Query;

import com.bsas.androiddevs.manejoerrores.repository.db.entity.MovieDb;

import java.util.List;

@Dao
public interface MovieDao extends BaseDao<MovieDb> {

    @Query("SELECT COUNT(*) FROM MOVIES")
    int countAll();

    @Query("SELECT * FROM MOVIES")
    List<MovieDb> findAll();
}

File: app/src/main/java/com/bsas/androiddevs/manejoerrores/repository/db/dao/MovieDao.kt
package com.bsas.androiddevs.manejoerrores.repository.db.dao

import androidx.room.Dao
import androidx.room.Query
import com.bsas.androiddevs.manejoerrores.repository.db.entity.MovieDb

@Dao
interface MovieDao : BaseDao<MovieDb> {

    @Query("SELECT COUNT(*) FROM MOVIES")
    suspend fun countAll(): Int

    @Query("SELECT * FROM MOVIES")
    suspend fun findAll(): List<MovieDb>

}